### ISSUE TITLE
Jackett is not working with Sonarr v3 if the tracker has SupportsImdbTVSearch

### ISSUE THREAD

**Issue by ngosang** [Jackett/Jackett#8107]
**TLDR;** Jackett is not working with Sonarr v3 if the tracker has `SupportsImdbTVSearch`.

I have been investigating how Sonarr v3 works (It's different to v2).

1. **If the tracker supports `SupportsImdbTVSearch`**
It sends a search with imdbid (from tv series, not episode) + season + episode => `?t=tvsearch&cat=5030,5040&extended=1&apikey=5k29tomyy11df265gio7iouhqx9k4&offset=0&limit=100&imdbid=tt2861424&season=4&ep=6`
* If the response has results, that's all. **The error is here, keep reading.**
* If the response doesn't has results, it try to do **string search** as a fallback => `?t=tvsearch&cat=5030,5040&extended=1&apikey=5k29tomyy11df265gio7iouhqx9k4&offset=0&limit=100&q=Rick%20et%20Morty&season=4&ep=6`.This is working fine.

2. **If the tracker doesn't supports `SupportsImdbTVSearch`**
It sends the string search as in previous case. This is working fine.

3. **Sonarr v2 is not using `SupportsImdbTVSearch`**
It sends the string search as in previous case. This is working fine.

**Problem with Jackett**
In most trackers (if not all) we are doing something like this:
```
if (query.IsImdbQuery)
      => Search by query.ImdbID
else
      => Search by query.GetQueryString()
```
This is fine for Movies because the IMDB id is unique, but is not working with tv series, because this will return all episodes (or the first 100 matches). **We are not using the season + episode parameters provided by Sonarr.**

The solution is complex because most tracker don't support searching IMDB + season/episode in the same query. It works only in some of them, for example, in IPTorrents you can search `tt2861424 S04E04`.

By now I think the best solution is disable `SupportsImdbTVSearch` in all trackers until we have a global solution. I don't want to do a "hack" for IPTorrents => #8102 because we will have this problem again in other trackers. If the tracker supports `SupportsImdbMovieSearch` Sonarr v3 still works and you can search movies and tv series in Jackett UI.

Related #4859 #7977

@Jackett/maintainers @RoloSoze

**Comment by RoloSoze** [comment]
From my perspective as someone who only uses Jackett, but not Sonarr, it's very unfortunate to have the IMDb ID search disabled for TV Shows, especially after all the work that you guys have done in #4859 

However, if it's only temporary until a solution is found, if there is currently no other way around it, and it's for the greater good, then I'm personally fine with it. 


Just for my own understanding, I need a quick clarification on something you mentioned:


>**We are not using the season + episode parameters provided by Sonarr.**

So will using these parameters help with the problem? 'Cause the sentence mentioned after that, 




> The solution is complex because most tracker don't support searching IMDB + season/episode in the same query.

makes me think that using these parameters will not work because of this limitation.

**Comment by ngosang** [comment]
> From my perspective as someone who only uses Jackett, but not Sonarr, it's very unfortunate to have the IMDb ID search disabled for TV Shows, especially after all the work that you guys have done in #4859

Se my last line, Jackett UI still works after this fix => ` If the tracker supports SupportsImdbMovieSearch Sonarr v3 still works and you can search movies and tv series in Jackett UI.`

> So will using these parameters help with the problem?

I want to hear more opinions before doing anything but my experience says me that we will have to provide a different solution for each tracker and in many of them won't be a fix available, even if you can search the by imdb tv series in the website.

**Comment by RoloSoze** [comment]
> See my last line, Jackett UI still works after this fix => ` If the tracker supports SupportsImdbMovieSearch Sonarr v3 still works and you can search movies and tv series in Jackett UI.`

I missed that part. Since there's no impact on Jackett users, I'm totally fine with whatever you guys decide.

**Comment by kevindd992002** [comment]
This is interesting and I think it is what's causing my issue. I'm also using Sonarr v3 with Jackett 0.14.476.0 and I noticed that I was not getting all TorrentDay (through Jackett) results when I tried downloading all episodes for the TV show "The Neighborhood" (IMDB ID: tt7942794). At first I thought it was bad indexing from TorrentDay but I highly doubt it. Then I tried searching using the Jackett webui and I noticed that it only returned 100 results. 

So how does Sonarr search using the IMDB ID plus season and episode and how does Jackett return the results? I tried searching "tt7942794 s01e16" and it's not working. What works is when I search "tt7942794" alone but like I said there are only 100 results.

**Comment by kevindd992002** [comment]
Interestingly enough, I tried searching "tt7942794 s01e16" in TorrentDay itself and it yielded the correct results. Here's the difference:

![image](https://user-images.githubusercontent.com/3907957/78921209-279e7c80-7ac7-11ea-8c27-470ebaa6464b.png)

![image](https://user-images.githubusercontent.com/3907957/78921242-2f5e2100-7ac7-11ea-9d02-852d716fef53.png)

So this tells me that the IMDB ID + SeasonEpisode format works for TorrentDay but Jackett is somehow not parsing it correctly? I'm not sure.

**Comment by cadatoiva** [comment]
I agree with the short term fix. 

A suggestion for the long term fix would be something to the effect of looking up the imdb id on a tv search, and collecting the appropriate season and episode imdb ids, search all three (show, season, and episode) and then filtering the results returned to the interface. Or at least some internal infrastructure that facilitates this for trackers where it helps.

Between this suggestion and the IMDB + Episode string query mentioned should cover most cases.

The more we can abstract away the work out of individual trackers, the better.

**Comment by jyggen** [comment]
I'll look into this more properly later, but a quick thought is that you can find exactly when and how Sonarr does its Torznab searching [here](https://github.com/Sonarr/Sonarr/blob/24fba7a36dc0ff5300a02522f3d0b61bb35aca7f/src/NzbDrone.Core/Indexers/Newznab/NewznabRequestGenerator.cs) - it basically boils down to what Jackett returns on the capabilities lookup. I think it'll be hard to abstract it away from individual trackers since each indexer's capabilities depends on how their search is implemented/behaves.

**Comment by kevindd992002** [comment]
How long after a commit gets merged to master will there be a release that contains that commit?

**Comment by garfield69** [comment]
Releases are currently published by hand. 
I publish pre-release on demand, and a stable release is usually published daily by myself around the same time of day, which recently is in about 16 hours from now, plus or minus an hour.
That can change depending on how much update activity the project is going through. This last month has been particularly active.

**Comment by kevindd992002** [comment]
I see. How do I tell if the release is considered as stable versus a development one though?

**Comment by cadatoiva** [comment]
@jyggen, what I mean by abstracting is laying down the heavy work that most indexers need, so they can focus only on implementing the specifics related to how their tracker needs the data. Similar to how we have the Cardigann processor, and that abstraction allows us to have a large number of indexers that have zero C# code in use through their YAML definitions.

**Comment by garfield69** [comment]
>How do I tell if the release is considered as stable versus a development one though?

The jackett pre-release are only auto-downloaded by jackett if you have ticked the **update to pre-release** checkbox on you jackett dashboard.
And on the github https://github.com/Jackett/Jackett/releases page those releases are tagged with the  **pre release** tag

**Comment by kevindd992002** [comment]
> 
> 
> > How do I tell if the release is considered as stable versus a development one though?
> 
> The jackett pre-release are only auto-downloaded by jackett if you have ticked the **update to pre-release** checkbox on you jackett dashboard.
> And on the github https://github.com/Jackett/Jackett/releases page those releases are tagged with the **pre release** tag

But if I use the LSIO Jackett docker container and want to be on the latest release, I need to be using the development tag, correct?

**Comment by garfield69** [comment]
>But if I use the LSIO Jackett docker container and want to be on the latest release, I need to be using the development tag, correct?

So to clarify, the pre-release is not a separate development stream, it is simply the master stream with a pre-release tag.
So when Jackett does an auto-update, and you are using the allow-update-to-pre-release enabled, Jackett will still fetch the latest stable if that version is newer than the previous version.

With docker its a little different.
The Jackett docker images are not built or maintained by us.
The LSIO team keep two separate streams, and when you pull docker with the `latest` tag you get the stable release of Jackett, and when you pull with the `development` tag you get the lastest pre-release.
So if you want to be on the latest version of Jackett you will want to pull from the `latest` tag, and only pull from the `development` tag when you are expecting a pre-release to solve a ticket you raised.

**Comment by kevindd992002** [comment]
> > But if I use the LSIO Jackett docker container and want to be on the latest release, I need to be using the development tag, correct?
> 
> So to clarify, the pre-release is not a separate development stream, it is simply the master stream with a pre-release tag.
> So when Jackett does an auto-update, and you are using the allow-update-to-pre-release enabled, Jackett will still fetch the latest stable if that version is newer than the previous version.
> 
> With docker its a little different.
> The Jackett docker images are not built or maintained by us.
> The LSIO team keep two separate streams, and when you pull docker with the `latest` tag you get the stable release of Jackett, and when you pull with the `development` tag you get the lastest pre-release.
> So if you want to be on the latest version of Jackett you will want to pull from the `latest` tag, and only pull from the `development` tag when you are expecting a pre-release to solve a ticket you raised.

I see. I have the auto-update option in the GUI unchecked (as suggestes when using docker containers) so we can ignore that part.

So if I understand correctly, the pre-release just gets released first than the stable release by a few hours and that's it? If I want to be able to be on the bleeding edge, I should always the development tag, correct?

**Comment by garfield69** [comment]
>So if I understand correctly, the pre-release just gets released first than the stable release by a few hours and that's it? If I want to be able to be on the bleeding edge, I should always the development tag, correct?

no. ;-)
as I said earlier, the pre-release is published on demand. by this, I mean, it does not get built unless I want to publish one.
You will be as up to date as can be if you stick with the stable release.

The only time the stable will be out of date is during the period of time when we have made some significant change to jackett core, and want only users that have the pre-release update enabled on their dashboard to test jackett.
during that time, I will not be publishing a stable until we are satisfied the pre-release has not caused any issues.
Once we are satisfied the pre-release is good, then I will resume publishing the stable again.

**Comment by kevindd992002** [comment]
Ok. It's really confusing, sorry :) It's just that based [here](https://github.com/linuxserver/docker-jackett/releases) , the latest release and the latest pre-release have the same set of changelogs. So what's even the difference between the two?

I mean, I understand in jackett's github itself the latest release and the pre-release are not the same. But not sure why it's like that with LSIO's.

**Comment by garfield69** [comment]
The Jackett Github master is the base source for both stable and pre-release.
We do not have a *development* branch of Jackett. there is just the **master** source.
when I publish a stable or pre-release on Jackett github, the docker automation pulls our source and prepares either a `latest` tagged image or a `development` image.
the docker's use of `development` is a misnomer. 

Look, don't get hung up about it.
my advice to you is to continue to pull from the docker `latest` images and leave it at that. It's all you need to know.

**Comment by kevindd992002** [comment]
Yeah, that's what I did. I was just curious on how github releases and docker container update automation integrate with each other, in general. Sorry if it's a bit off-topic but thanks for the information so far.

**Comment by jyggen** [comment]
> 
> 
> Yeah, that's what I did. I was just curious on how github releases and docker container update automation integrate with each other, in general. Sorry if it's a bit off-topic but thanks for the information so far.

I think you'll get a much better answer if you ask the LSIO guys directly since they're the ones doing the Docker container and its releases :)

**Comment by SunsparcSolaris** [comment]
Updated to x.501 docker, confirmed working. Thanks everyone!

**Comment by morpheus133** [comment]
Hi!
TVstore is working with imdbtv support! I was the one who ask merging the IMDBTV support and the first indexer what was supporting it was tvdb.
The solution is there that jackett get result for all episode and season and parse it based on the incomming season and episode, and send back only the good one. (The same methode is used by ncore. (If it will be merged))

**Comment by garfield69** [comment]
yes, I think we can switch tvstore back on, since that indexer was designed from the beginning to handle how the search-tv imdbid was set up.
Its the other indexers that we really need to switch off, since that we discovered they all need the tvstore logic to work properly with this feature enabled.
So rather than have to insert multiple duplicate code in every indexer, the team is looking to add a support module that does what tvstore does for imdbid episodes, but which will allow other indexers to also benefit.
At least, that is what I think may happen, I could be wrong. 
anyway, i'll flip the switch on tvstore.

**Comment by cadatoiva** [comment]
> So rather than have to insert multiple duplicate code in every indexer, the team is looking to add a support module that does what tvstore does for imdbid episodes, but which will allow other indexers to also benefit.
> At least, that is what I think may happen, I could be wrong.

That's the goal at least. As far as making it happen, that's another story completely.

**Comment by morpheus133** [comment]
If you touch this part some kind of omdb fallback can be also added. If jackett dont find anything with imdbid it can lookup omdb for title and make a search with it also. ( Currently omdb apikey not visible by parsers.)

**Comment by garfield69** [comment]
Jackett v0.14.541 for tvstore reactivation

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/pull/8108#issuecomment-612506088

**Comment by kevindd992002** [comment]
Would you guys be willing to turn back on the TorrentDay IMDB ID search and fix the search bug for me?

TorrentDay's search in their website works when using "imdb id + SxxExx" but not when searched through Jackett's UI.

**Comment by kevindd992002** [comment]
> 
> 
> Would you guys be willing to turn back on the TorrentDay IMDB ID search and fix the search bug for me?
> 
> TorrentDay's search in their website works when using "imdb id + SxxExx" but not when searched through Jackett's UI.

Anybody can help me with this?

**Comment by garfield69** [comment]
>TorrentDay's search in their website works when using "imdb id + SxxExx" but not when searched through Jackett's UI.
>Anybody can help me with this?

devs are working on other tickers older than this one, so no, no one can spare the time for this right now.
we will be in touch when resources become available.

**Comment by kevindd992002** [comment]
> 
> 
> > TorrentDay's search in their website works when using "imdb id + SxxExx" but not when searched through Jackett's UI.
> > Anybody can help me with this?
> 
> devs are working on other tickers older than this one, so no, no one can spare the time for this right now.
> we will be in touch when resources become available.

Thanks for the confirmation.
